scaleway/resource_redis_cluster_test.go

```diff
@@ -350,3 +600,63 @@ func testAccCheckScalewayRedisExists(tt *TestTools, n string) resource.TestCheck
 		return nil
 	}
 }
+
+func testAccCheckScalewayRedisPrivateNetworksIpsAreEither(name string, possibilities ...string) resource.TestCheckFunc {
+	return func(state *terraform.State) error {
+		rs, ok := state.RootModule().Resources[name]
+		if !ok {
+			return fmt.Errorf("resource not found: %s", name)
+		}
+		actualIPs := []string(nil)
+		for i := range possibilities {
+			actualIPs = append(actualIPs, rs.Primary.Attributes[fmt.Sprintf("private_network.%d.service_ips.0", i)])
+		}
+		for _, ip := range actualIPs {
+			for i := range possibilities {
+				if possibilities[i] == ip {
+					possibilities[i] = "ip found"
+				}
+			}
+		}
+		for _, p := range possibilities {
+			if p != "ip found" {
+				return fmt.Errorf("no attribute private_network.*.service_ips.0 was found with value %v", p)
+			}
+		}
+		return nil
+	}
+}
+
+func testAccCheckScalewayRedisPrivateNetworksIdsAreEither(name string, possibilities ...string) resource.TestCheckFunc {
```

Comment:
Interesting function I wonder if they could be integrated in: https://github.com/hashicorp/terraform-plugin-sdk/blob/v2.17.0/helper/resource/testing.go